spec/query_spec.cr

```diff
@@ -661,5 +661,31 @@ describe Avram::Query do
       original_query.first.should_not eq nil
       new_query.should eq 1
     end
+
+    it "clones preloads" do
+      Avram.temp_config(lazy_load_enabled: false) do
+        post = PostBox.create
+        comment = CommentBox.create &.post_id(post.id)
+        posts = Post::BaseQuery.new.preload_comments
+        cloned_posts = posts.clone
+        posts.first.comments.should eq([comment])
+        cloned_posts.first.comments.should eq([comment])
+      end
+    end
+
+    it "clones nested preloads" do
+      Avram.temp_config(lazy_load_enabled: false) do
+        item = LineItemBox.create
+        price = PriceBox.create &.line_item_id(item.id).in_cents(500)
+        product = ProductBox.create
+        LineItemProductBox.create &.line_item_id(item.id).product_id(product.id)
+
+        products = Product::BaseQuery.new
+          .preload_line_items(LineItem::BaseQuery.new.preload_price)
+        cloned_products = products.clone
+        products.first.line_items.first.price.as(Price).in_cents.should eq(500)
+        cloned_products.first.line_items.first.price.as(Price).in_cents.should eq(500)
```

Comment:
I think you could shorten this up by doing something like `line_items.first.price.should_not be_nil`